Library/Homebrew/keg_relocate.rb

```diff
@@ -87,15 +87,21 @@ def change_rpath(file, new_prefix)
     end
     return unless patchelf.installed?
     cmd = "#{patchelf.bin}/patchelf --set-rpath #{new_prefix}/lib"
+    test_cmd = "#{patchelf.bin}/patchelf --print-rpath --print-interpreter #{file}"
+    (old_i, old_p) = `#{test_cmd}`.split(/\n/)
     if file.mach_o_executable?
       interpreter = new_prefix == PREFIX_PLACEHOLDER ?
         "/lib64/ld-linux-x86-64.so.2" :
         HOMEBREW_PREFIX/"lib/ld.so"
-      cmd << " --set-interpreter #{interpreter}"
+      cmd << " --set-interpreter #{interpreter}" unless old_i == interpreter
     end
     cmd << " #{file}"
-    puts "Setting RPATH of #{file}" if ARGV.debug?
-    safe_system cmd
+    if old_p == "#{new_prefix}/lib"
```

Comment:
There's the situation that if the rpath is right but the interpreter is wrong, it won't get fixed up.
Can you change this logic to:
```ruby
if old_p == "#{new_prefix}/lib" && old_i == interpreter
```